packages/signature-controller/src/types.ts

```diff
@@ -71,15 +77,41 @@ export type MessageParamsTyped = MessageParams & {
         primaryType: string;
         message: Json;
       };
-
   /** Version of the signTypedData request. */
   version?: string;
 };
 
+/** Information about various state changes returned by decoding api. */
+type DecodingDataStateChanges = {
+  assetType: string;
+  changeType: string;
+  address: string;
+  amount: string;
+  contractAddress: string;
```

Comment:
Should we use the `Hex` type for some of these?

Reply:
According to api [doc](https://www.notion.so/metamask-consensys/Signature-API-Documentation-fb31344c77454671ad427c5b88f06b84) it is string types.